File: src/main/java/toast/pine/Commands/addFriend.java
package toast.pine.Commands;

import org.bukkit.entity.Player;
import org.bukkit.persistence.PersistentDataContainer;
import revxrsal.commands.annotation.Command;
import revxrsal.commands.annotation.Default;
import toast.pine.Keys;
import toast.pine.SocialSystem.PlayerSocial;
import toast.pine.ToastRPG;

public class addFriend {


    @Command({"addfriend", "af", "addf"})
    public void addFriend(Player stranger, @Default("self") Player sender) {
        PersistentDataContainer strangerPDC = stranger.getPersistentDataContainer();
        PersistentDataContainer senderPDC = sender.getPersistentDataContainer();

        PlayerSocial strangerSocial = strangerPDC.get(Keys.SOCIAL_PROFILE, ToastRPG.getAdapterManager().getSocialProfileAdapter());
        PlayerSocial senderSocial = senderPDC.get(Keys.SOCIAL_PROFILE, ToastRPG.getAdapterManager().getSocialProfileAdapter());

        senderSocial.getFriends().sendInvite(strangerSocial);
    }

}

File: src/main/java/pine/toast/Commands/AddFriend.kt
package toast.pine.commands

import org.bukkit.entity.Player
import org.bukkit.persistence.PersistentDataContainer
import revxrsal.commands.annotation.Command
import revxrsal.commands.annotation.Default
import pine.toast.Keys
import toast.pine.socialSystem.PlayerSocial
import pine.toast.ToastRPG

class AddFriend {


    @Command("addfriend", "af", "addf")
    fun addFriend(stranger: Player, @Default("self") sender: Player) {
        val strangerPDC: PersistentDataContainer = stranger.persistentDataContainer
        val senderPDC: PersistentDataContainer = sender.persistentDataContainer
        val strangerSocial: PlayerSocial = strangerPDC.get(Keys.SOCIAL_PROFILE, ToastRPG.getAdapterManager()!!.socialProfileAdapter)!!
        val senderSocial: PlayerSocial = senderPDC.get(Keys.SOCIAL_PROFILE, ToastRPG.getAdapterManager()!!.socialProfileAdapter)!!
        senderSocial.getFriends().sendInvite(strangerSocial)

    }
}
